sync/bi_sync_mysql.go

```diff
@@ -25,59 +27,105 @@ import (
 )
 
 var (
+	GTID            string
 	dataSourceName1 string
 	dataSourceName2 string
+	engin           *xorm.Engine
 	engin1          *xorm.Engine
 	engin2          *xorm.Engine
+	serverId1       uint32
+	serverId2       uint32
+	serverUUID1     string
+	serverUUID2     string
 )
 
-func InitConfig() *canal.Config {
+func InitConfig() (*canal.Config, *canal.Config) {
 	// init dataSource
 	dataSourceName1 = fmt.Sprintf("%s:%s@tcp(%s:%d)/%s", username1, password1, host1, port1, database1)
 	dataSourceName2 = fmt.Sprintf("%s:%s@tcp(%s:%d)/%s", username2, password2, host2, port2, database2)
 
 	// create engine
 	engin1, _ = CreateEngine(dataSourceName1)
 	engin2, _ = CreateEngine(dataSourceName2)
-	log.Info("init engine success…")
 
-	// config canal
-	cfg := canal.NewDefaultConfig()
-	cfg.Addr = fmt.Sprintf("%s:%d", host1, port1)
-	cfg.Password = password1
-	cfg.User = username1
+	// get serverId
+	serverId1, _ = GetServerId(engin1)
+	serverId2, _ = GetServerId(engin2)
+
+	// get server uuid
+	serverUUID1, _ = GetServerUUID(engin1)
+	serverUUID2, _ = GetServerUUID(engin2)
+
+	// config canal1
+	cfg1 := canal.NewDefaultConfig()
+	cfg1.Addr = fmt.Sprintf("%s:%d", host1, port1)
+	cfg1.Password = password1
+	cfg1.User = username1
 	// We only care table in database1
-	cfg.Dump.TableDB = database1
+	cfg1.Dump.TableDB = database1
+
+	// config canal2
+	cfg2 := canal.NewDefaultConfig()
+	cfg2.Addr = fmt.Sprintf("%s:%d", host2, port2)
+	cfg2.Password = password2
+	cfg2.User = username2
+	// We only care table in database2
+	cfg2.Dump.TableDB = database2
 	// cfg.Dump.Tables = []string{"user"}
 	log.Info("config canal success…")
-	return cfg
+	return cfg1, cfg2
 }
 
 func StartBinlogSync() error {
+	var wg sync.WaitGroup
+
 	// init config
-	config := InitConfig()
+	config1, config2 := InitConfig()
 
-	c, err := canal.NewCanal(config)
-	pos, err := c.GetMasterPos()
+	c1, err := canal.NewCanal(config1)
+	GTIDSet1, err := c1.GetMasterGTIDSet()
 	if err != nil {
 		return err
 	}
 
 	// Register a handler to handle RowsEvent
-	c.SetEventHandler(&MyEventHandler{})
+	c1.SetEventHandler(&MyEventHandler{})
+
+	// Start canal2
+	go func() {
+		err := c1.StartFromGTID(GTIDSet1)
+		if err != nil {
+			panic(err)
+		}
+	}()
+	wg.Add(1)
 
-	// Start canal
-	c.RunFrom(pos)
+	c2, err := canal.NewCanal(config2)
+	GTIDSet2, _ := c2.GetMasterGTIDSet()
 
+	// Register a handler to handle RowsEvent
+	c2.SetEventHandler(&MyEventHandler{})
+
+	// Start canal2
+	go func() {
+		err := c2.StartFromGTID(GTIDSet2)
+		if err != nil {
+			panic(err)
+		}
+	}()
+	wg.Add(1)
+
+	wg.Wait()
 	return nil
 }
 
 type MyEventHandler struct {
```

Comment:
Define `Database` struct, put dataSourceName, engine, serverId, serverUUID into it as members